Модуль 5. Repositories

artemvolt opened the pull request:
- Добавил модели для таблиц.
- Добавил репозитории.
- Добавил тесты.

artemvolt committed:
Модуль 5. Repositories

- Добавил модели для таблиц.
- Добавил репозитории.
- Добавил тесты.

artemvolt committed:
Модуль 5. Repositories

- Поправил аннотации к интерфейсу.

AristarkhLetskin commented on common/Infrastructure/Adapters/Postgres/Repositories/CourierRepository.php:
```diff
@@ -0,0 +1,92 @@
+        }
+    }
+
+    public function updateCourier(CourierAggregate $courier): void
```
В рамках сценария подразумевается все таки, что аггрегат сначала был восстановлен и поэтому кейса что его нет не должно быть (плюс получается тут еще доп запрос.
Плюс скорее всего ORM в рамках запроса при `->save();` строит запрос с `where id = courierId` и этого вполне достаточно для уверенности апдейта

artemvolt replied:
К сожалению, в фреймворке, в yii2 при вызове метода save, он создаст новую запись в бд, если она не была создана, поэтому приходится делать доп проверку.
Я решил не завязывать доменные сущности на orm, чтобы проще было потом с ними работать.

AristarkhLetskin commented on common/Infrastructure/Adapters/Postgres/Repositories/CourierRepository.php:
```diff
@@ -0,0 +1,92 @@
+    public function getById(UuidInterface $id): CourierAggregate
+    {
+        $found = CourierModel::findOne(['id' => $id->toString()]);
+        if (null === $found) {
```
тут лучше не выкидывать исключение, а сделать сигнатуру метода как nullable reference, то есть `?CourierAggregate`, это даст возможность проверить вызывающему коду есть ли курьер или нет и уже выстраивать логику (доменную или приложения) от этого

artemvolt replied:
Исправил.

artemvolt committed:
Модуль 5. Repositories

- Добавил todo на исправление тестов.
- Сделал возврат null, если сущность не найдена в методе getById.
